File: packages/java/sdk/src/main/java/dev/runfabric/Handler.java
package dev.runfabric;

import java.util.Map;

/**
 * RunFabric handler contract: (event, context) -> response map.
 */
@FunctionalInterface
public interface Handler {
    /**
     * Handle an invocation event.
     *
     * @param event   event payload (e.g. JSON body as map)
     * @param context request context (stage, function name, request ID)
     * @return response map (serialized as JSON by the runtime)
     */
    Map<String, Object> handle(Map<String, Object> event, HandlerContext context);
}

File: packages/dotnet/sdk/RunFabric.Sdk/Handler.cs
namespace RunFabric.Sdk;

/// <summary>RunFabric handler contract: (event, context) -> response dictionary.</summary>
/// <param name="event">Event payload (e.g. JSON body as dictionary).</param>
/// <param name="context">Request context (stage, function name, request ID).</param>
/// <returns>Response dictionary (serialized as JSON by the runtime).</returns>
public delegate IReadOnlyDictionary<string, object?> Handler(
    IReadOnlyDictionary<string, object?> event,
    HandlerContext context);
